### DISCUSSION TITLE
About RoiCenterCrop

### DISCUSSION THREAD

**Discussion by XieXiao1997** [Project-MONAI/MONAI#3273] [Q&A] [answered]
In my project, the ROI is bilateral asymmetric, similar to the breast area, and the ROI class is the same "1". 
What method should I use if I want to cut both sides with ROI as the center? I tried using the SpatialCrop method, but it is not supported to input ROI images directly for image clipping.

**Comment by Nic-Ma** [comment]
Hi @ericspod @dongyang0122 @wyli ,

I feel this question is related to some research discussion, could you please help share some experience here?

Thanks in advance.

**Comment by ericspod** [comment, marked as answer]
Perhaps [CropForeground](https://github.com/Project-MONAI/MONAI/blob/856237e28aff1465bd55e9c7eb22345bcc664cf4/monai/transforms/croppad/array.py#L655 ) is what you want. You can provide a callable to choose another criteria if you want, by default it's `is_positive` based on an input label in a particular channel. If you're using dictionary based transforms `CropForegroundD` is what you'd want. Docs here: https://docs.monai.io/en/latest/transforms.html#cropforeground

**Reply by XieXiao1997** [reply]
I used Kmeans to solve the problem and itthen used the SpatialCrop based cluster center.. it worked

**Reply by ericspod** [reply]
Ok so your solution was to use a separate technique to find a ROI center and use that. It works with `SpatialCrop` but also with `CropForeground` by using your technique in a function passed in as `select_fn`.
